src/main/java/org/broadinstitute/consent/http/mail/SendGridAPI.java

```diff
@@ -69,7 +122,13 @@ private Optional<Response> sendMessage(Mail message) {
                 for (String key : sendGrid.getRequestHeaders().keySet())
                     request.addHeader(key, sendGrid.getRequestHeaders().get(key));
                 // send
-                return Optional.of(sendGrid.makeCall(request));
+                Response response = sendGrid.makeCall(request);
+                if (response.getStatusCode() > 202) {
```

Comment:
Silly question, but why not log all messages sent?  I could see where you might not want them all as errors, but I thought I read that one of the concerns that there were missing messages.  By logging the response for all messages might give us insight into if DUOS sent the message but SendGrid lost it.

Reply:
In general, we don't do db pruning. I can envision a targeted event where we might need to do that, but as a matter of application functionality, we tend not to go down that road.

Reply:
Docs for what I asserted above: https://docs.sendgrid.com/api-reference/mail-send/mail-send#responses